include/mata/nfa/delta.hh

```diff
@@ -366,68 +409,75 @@
      */
     void add(const State state_from, const Symbol symbol, const StateSet& states);
 
+    using const_iterator = std::vector<StatePost>::const_iterator;
+    const_iterator cbegin() const { return state_posts_.cbegin(); }
+    const_iterator cend() const { return state_posts_.cend(); }
+    const_iterator begin() const { return state_posts_.begin(); }
+    const_iterator end() const { return state_posts_.end(); }
+
     /**
-     * Iterator over transitions. It iterates over triples (lhs, symbol, rhs) where lhs and rhs are states.
+     * @brief Iterator over transitions represented as @c Transition instances.
+     *
+     * It iterates over triples (source, symbol, target).
      */
-    struct transitions_const_iterator {
-    private:
-        const std::vector<StatePost>& post_;
-        size_t current_state_;
-        StatePost::const_iterator post_iterator_{};
-        StateSet::const_iterator targets_position_{};
-        bool is_end_;
-        Transition transition_{};
-
+    class Transitions {
```

Comment:
Hm, I am still wandering whether having all this code just to avoid one for loop when printing out automaton is worth it ... I know that it is stupid to say this after it is done.

Reply:
In that case, what about the class `Transition` and functions accepting this class? We would have no way to create an instance of `Transition`, so then the whole class would be pointless and could be removed, too, no?

Reply:
I think having an iterator over all transitions can be useful in some other contexts than printing out automaton. For example if I want to check if automaton accepts only one symbol words, I can iterate all transitions and check if source state is initial and target is final (I feel like we did something like that in Noodler).

Reply:
ok, lets keep it